src/api-v1.js

```diff
@@ -580,6 +580,105 @@ api.declare({
 
 api.declare({
   method: 'put',
+  route: '/ami-set/:amiSetId',
+  name: 'createAmiSet',
+  deferAuth: true,
+  scopes: [['aws-provisioner:manage-ami-set:<amiSetId>']],
+  input: 'create-ami-set-request.json#',
+  output: 'get-ami-set-response.json#',
+  title: 'Create new AMI Set',
+  stability:  base.API.stability.stable,
+  description: [
+    'Create an AMI Set. An AMI Set is a collection of AMIs with a single name.',
+  ].join('\n'),
+}, async function (req, res) {
+  let input = req.body;
+  let amiSet = req.params.amiSetId;
+
+  input.lastModified = new Date();
+
+  // Authenticate request with parameterized scope
+  if (!req.satisfies({amiSet: amiSetId})) {
+    return;
+  }
+
+  // Create amiSet
+  let aSet;
+  try {
+    aSet = await this.amiSet.create(amiSet, input);
+  } catch (err) {
+    // We only catch EntityAlreadyExists errors
+    if (!err || err.code !== 'EntityAlreadyExists') {
+      throw err;
+    }
+    aSet = await this.amiSet.load({amiSetId});
+
+    // Check if it matches the existing amiSet
+    let match = [
+      'amiSetId',
+      'amis',
+    ].every((key) => {
+      return _.isEqual(aSet[key], input[key]);
+    });
+
+    // If we don't have a match we return 409, otherwise we continue as this is
+    // is an idempotent operation.
+    if (!match) {
+      res.status(409).json({
+        error: 'AMI Set already exists with different definition',
+      });
+      return;
+    }
+  }
+
+  // Publish pulse message
+  await this.publisher.amiSetCreated({
+    amiSet: amiSetId,
+  });
```

Comment:
We don't actually consume any pulse messages in the provisioner.  We scan the worker type table on each iteration, since we need to load all the worker types anyway to see what the max/min capacity, scaling ratio and instance types.  We could maintain that state in the backend process, but that's not there yet and would be a fairly substantial code change.  I don't know if the Azure overhead (tech or billing wise) is worth the trouble.

That said, I don't think that we really need to have pulse messages for AmiSets.  The important Pulse messages are for WorkerType updates, since those are (well, should be) used to gracefully shutdown old instances when we update the worker type definition.